moveit_commander/src/moveit_commander/planning_scene_interface.py

```diff
@@ -167,23 +109,23 @@ def add_plane(self, name, pose, normal = (0, 0, 1), offset = 0):
         p.coef.append(offset)
         co.planes = [p]
         co.plane_poses = [pose.pose]
-        self._pub_co.publish(co)
-        
-    def attach_mesh(self, link, name, pose = None, filename = '', size = (1, 1, 1), touch_links = []):
+        self.__submit(co, attach=False)
+
+    def attach_mesh(self, link, name, pose=None, filename='', size=(1, 1, 1), touch_links=[]):
         aco = AttachedCollisionObject()
-        if pose!=None and filename:
+        if (pose is not None) and filename:
             aco.object = self.__make_mesh(name, pose, filename, size)
         else:
             aco.object = self.__make_existing(name)
         aco.link_name = link
         aco.touch_links = [link]
         if len(touch_links) > 0:
             aco.touch_links = touch_links
-        self._pub_aco.publish(aco)
+        self.__submit(aco, attach=True)
 
-    def attach_box(self, link, name, pose = None, size = (1, 1, 1), touch_links = []):
+    def attach_box(self, link, name, pose=None, size=(1, 1, 1), touch_links=[]):
         aco = AttachedCollisionObject()
-        if pose!=None:
+        if pose is not None:
```

Comment:
You shouldn't have to double negate. `if pose:` should be enough.

Reply:
No. There is a semantic difference between
`if pose:`
and
`if pose is not None:`
The latter compares against the `None` object, while the former considers any truth value of `pose`.

Reply:
Agree with @rhaschke on this. Here it might not matter, but the Pose class could do something like override `__bool__()` to return `False` for poses where all elements are zero (it doesn't but it could). So it's actually more correct to say `pose is not None`